src/ipa_pc/mod.rs

```diff
@@ -524,6 +492,12 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         Ok(pp)
     }
 
+    fn setup_oracle() -> Self::RandomOracle {
```

Comment:
I've placed this function here because of 2 reasons:

1. PROTOCOL_NAME is used as seed, and this constant is a part of InnerProductArgPC struct

2. FiatShamirRng requires "D: Digest" bound, which is present in InnerProductArgPC, but not present in general PolynomialCommitment trait

So the setup call is laconic:

let fs_rng = PC::setup_oracle()

If we remove this function, everywhere when we need to spawn fs_rng we also need to do something like:

use blake2::Blake2s;
use crate::rng::{FiatShamirRng, FiatShamirChaChaRng};

...

let fs_rng = FiatShamirChaChaRng::<Blake2s>::from_seed(
      &to_bytes![&PC::PROTOCOL_NAME].unwrap(),
);

Or maybe there is some other elegant solution, which I don't see now?

Reply:
The setup string is different in Marlin with respect to poly-commit.
I wanted to keep it simple for now, just by introducing a RandomOracle type in PolynomialCommitment trait (without any specification of the traits this type should implement); in IPA trait you specify RandomOracle = FiatShamirChaChaRng<D> and in Marlin you explicitly instantiate a FiatShamirChaChaRng<D> and pass it around. 
If you want to keep with this trait (it's perfectly fine, it's something later we would've done anyway), let's move the setup() function in the trait and let it take a byte array as input (so that Marlin and IPA are allowed to choose their own separate initialization string), and maybe let's add too a function
`squeeze_128_bits_challenge<F: Field>(&mut self) -> F`
That will be simply a wrapper for the usual way to get a challenge:
`return u128::rand(self).into()`

Reply:
Done